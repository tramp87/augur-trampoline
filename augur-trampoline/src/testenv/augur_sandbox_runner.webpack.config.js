// @flow

// eslint-disable-next-line import/no-nodejs-modules,import/no-commonjs
const path = require('path');

// eslint-disable-next-line import/no-commonjs
module.exports = {
  entry: './src/testenv/augur_sandbox_runner.js',
  output: {
    path: path.resolve(__dirname, '../../dev-artifacts'),
    filename: 'augur_sandbox_runner.js',
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        exclude: /(node_modules|bower_components)/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: ['@babel/preset-env'],
            plugins: ['@babel/plugin-transform-runtime'],
          },
        },
      },
    ],
  },
  node: {
    fs: 'empty',
  },
};
