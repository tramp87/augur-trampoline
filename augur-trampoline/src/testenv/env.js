// @flow

import fetchPonyfill from 'fetch-ponyfill';
import ethJsUtil from 'ethereumjs-util';

const { fetch } = fetchPonyfill({});

const TESTRPC_HTTP_URL = 'http://ganache:8545';
const TESTRPC_WS_URL = 'ws://ganache:8545';
const AUGUR_CONTRACTS_URL = 'http://testenv';
const AUGUR_CONTRACTS_BROWSER_URL = '/api/contracts';

async function getContractAddresses(isClientSide: boolean): Promise<*> {
  const response = await fetch(
    isClientSide ? AUGUR_CONTRACTS_BROWSER_URL : AUGUR_CONTRACTS_URL,
  );
  const addresses = await response.json();
  return addresses;
}

const account = {
  signer: new Buffer(
    ethJsUtil.toBuffer(
      '0x73dff7a656b0ecc3bb281bd5d14f9f8e77b60355d6274683d2f6fc5e3ab7ac11',
    ),
  ),
  accountType: 'privateKey',
};

export {
  TESTRPC_HTTP_URL,
  TESTRPC_WS_URL,
  AUGUR_CONTRACTS_URL,
  getContractAddresses,
  account,
};
