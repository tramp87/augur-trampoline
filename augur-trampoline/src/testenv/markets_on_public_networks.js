// @flow

const MARKETS_ON_PUBLIC_NETWORKS: * = {
  '4': {
    cfk_jail: {
      market: '0x62643d0b946899bf7770f9b15cf9ae6c73d2cf9d',
      transaction:
        '0xe51ff13a07d63acca787927938a1d0315a2ec0c513bd1bcdc62d950388141b60',
    },
    trump_impeachment: {
      market: '0x4c5c33c2d785c0d62f396cb420ec10050d581d08',
      transaction:
        '0x6af28ceaa3e4c10a4876dc1f502f25d10413f58e93df88ccffa6bd7016255aeb',
    },
    china_antibiotics: {
      market: '0x448d1d8280844513c38b2b73bde8a539d0022954',
      transaction:
        '0xdf446661b6c88ccac0017322cf9737425e086a12261ae3a86d64579de1491c8d',
    },
  },
  '1': {
    eth500: {
      market: '0xd2b4906276b6ed334604f914306158c05a92e41f',
      transaction:
        '0x1a502e0bcbcc6974c81164575539beb6a4049dfe347d179b9a266f8a8c7c6786',
    },
    rep32: {
      market: '0x4e0ee58bf4230d3b799584e1a6027b4bda9fc4ed',
      transaction:
        '0x364d9e4513b3c0de0f346014b29b025491c27431cb6fdab2ec978ef88a530609',
    },
    btc20k: {
      market: '0x71fe9c2ee36374a380347d205acea2ee5359325a',
      transaction:
        '0x1aa53f7b0e099855aeb17bb0f0b9e9b83145022df7e2771f5c4b70fcdb820e41',
    },
    eth_options_categorical: {
      market: '0xcea0b692b0f97323518478c037c8c9b37e8c3f60',
      transaction:
        '0xfb704f9d6cf74114342f27f96c5e71d25a4c405210c8917bbcf53798487d43d0',
    },
    trump_reelection_2020: {
      market: '0xdecbd869eccac116193886c3f7fa4a150ffab681',
      transaction:
        '0xfc27c84df6cb4810762c0e931051dc55c8531a14081760aea3168c39e034f220',
    },
    ethcall500: {
      market: '0xf3a4e2fc480ea4bc91c3313d97fc36d4d70bf452',
      transaction:
        '0x9a9c4534606ab62885e9420f4510327a0cf8b0d50979718b07ecc4b51d265273',
    },
    usd100scalar: {
      market: '0x3e4f7b408f7b5c103f94c7cd2b72b1073b0152c6',
      transaction:
        '0xe849de5d52c5f10c533004096bde9a2b9d5ac109266b25600d5b96e09b550994',
    },
  },
};

export default MARKETS_ON_PUBLIC_NETWORKS;
