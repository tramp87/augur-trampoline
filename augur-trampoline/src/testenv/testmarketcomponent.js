// @flow

import React, { Fragment } from 'react';
import Web3 from 'web3';
import nullthrows from 'nullthrows';
import BigNumber from 'bignumber.js';
import { Range as ImmRange } from 'immutable';
import fetchMarketData from '../steps/DisplayMarketData/fetch';
import type { MarketData } from '../steps/DisplayMarketData/fetch';

type Props = {|
  network: string,
  name: string,
  id: string,
  creationTX: string,
|};
type WrappedMarketData =
  | {|
      status: 'error',
      error: string,
    |}
  | {|
      status: 'ok',
      data: MarketData,
    |};
type State = { data: ?WrappedMarketData };

const Outcome = ({
  marketType,
  outcomes,
  index,
}: {
  marketType: BigNumber,
  outcomes: Array<string>,
  index: number,
}) => {
  if (marketType.toNumber() === 0) {
    // binary
    return nullthrows(['NO', 'YES'][index]);
  } else if (marketType.toNumber() === 1) {
    // categorical
    return nullthrows(outcomes[index]);
  } else if (marketType.toNumber() === 2) {
    // scalar
    // TODO: think of boundaries
    return nullthrows(['DOWN', 'UP'][index]);
  }

  throw new Error(`Unknown market type ${marketType}`);
};

class TestMarketDetails extends React.Component<Props, State> {
  state: State;
  _mounted: boolean;

  constructor(props: Props) {
    super(props);
    this.state = { data: null };
    this._mounted = false;
  }

  componentDidMount() {
    this._mounted = true;

    this.run().then(data => {
      if (this._mounted) {
        this.setState({ data });
      }
    });
  }

  async run(): Promise<WrappedMarketData> {
    try {
      const web3 = new Web3(window.web3.currentProvider);
      const data = await fetchMarketData(
        web3,
        this.props.id,
        this.props.creationTX,
      );
      return {
        status: 'ok',
        data,
      };
    } catch (e) {
      console.error(e);
      return {
        status: 'error',
        error: e.toString(),
      };
    }
  }

  componentWillUnmount() {
    this._mounted = false;
  }

  render() {
    const wrapped = this.state.data;

    const getDetails = () => {
      if (wrapped == null) {
        return (
          <span>
            Loading {this.props.id}
            ...
          </span>
        );
      }

      if (wrapped.status === 'error') {
        return (
          <span>
            Failed to fetch market {this.props.id}: {wrapped.error}
          </span>
        );
      }

      const data = wrapped.data;

      // TODO: this should come from some Augur.js enum
      const market_type_name = ['binary', 'categorical', 'scalar'];

      return (
        <span>
          [type:
          {nullthrows(market_type_name[data.marketType.toNumber()])}] [outcomes:
          {data.numberOfOutcomes.toString()}] {data.description}
          <ol>
            <li>id: {this.props.id}</li>
            <li>creation TX: {this.props.creationTX}</li>
            <li>debug: {JSON.stringify(data)}</li>
            <li>
              Outcomes:
              <ol>
                {ImmRange(0, data.numberOfOutcomes.toNumber())
                  .map(index => (
                    <li key={index}>
                      [0x
                      {index.toString(16)}]{' '}
                      <Outcome
                        marketType={data.marketType}
                        outcomes={data.outcomes}
                        index={index}
                      />
                    </li>
                  ))
                  .toArray()}
              </ol>
            </li>
          </ol>
        </span>
      );
    };

    return (
      <Fragment>
        [alias:
        {this.props.name}] {getDetails()}
      </Fragment>
    );
  }
}

export default TestMarketDetails;
