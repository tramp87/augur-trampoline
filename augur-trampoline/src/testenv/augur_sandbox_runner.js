// @flow

// we run Augur inside JSDOM to ensure clean shutdown

import nullthrows from 'nullthrows';
import Augur from 'augur.js';
import { TESTRPC_WS_URL } from './env';

async function run(func: *, params: *): Promise<*> {
  const augur = new Augur();
  await new Promise((resolve, reject) =>
    augur.connect(
      {
        ethereumNode: { ws: TESTRPC_WS_URL },
        augurNode: null,
      },
      (err, connectionInfo) =>
        err != null ? reject(err) : resolve(connectionInfo),
    ),
  );
  return await new Promise((resolve, reject) =>
    func(augur)({
      ...params,
      onSuccess: result => resolve(result),
      onFailed: e => reject(e),
    }),
  );
}

window.AUGUR_SANDBOX_RUNNER_PROMISE = run(
  nullthrows(window.AUGUR_SANDBOX_RUNNER_FUNC),
  nullthrows(window.AUGUR_SANDBOX_RUNNER_PARAMS),
);
