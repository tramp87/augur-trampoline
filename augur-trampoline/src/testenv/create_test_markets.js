// @flow

// eslint-disable-next-line import/no-nodejs-modules
import fs from 'fs';
import Web3 from 'web3';
import nullthrows from 'nullthrows';
import { JSDOM, VirtualConsole } from 'jsdom';
import { TESTRPC_HTTP_URL, getContractAddresses, account } from './env';

async function create_test_markets(): Promise<*> {
  const web3 = new Web3(new Web3.providers.HttpProvider(TESTRPC_HTTP_URL));
  const coinbase = await new Promise((resolve, reject) =>
    web3.eth.getCoinbase(
      (error, result) => (error != null ? reject(error) : resolve(result)),
    ),
  );
  const network = await new Promise((resolve, reject) =>
    web3.version.getNetwork(
      (error, result) => (error != null ? reject(error) : resolve(result)),
    ),
  );

  const addresses = await getContractAddresses().then(addresses =>
    nullthrows(addresses[network]),
  );

  const result = await runAugurInSandbox(
    augur => params => augur.api.Universe.createYesNoMarket(params),
    {
      _endTime: Math.floor(Date.now() / 1000 + 86400 * 100),
      _feePerEthInWei: '42',
      _denominationToken: addresses.Cash,
      _designatedReporterAddress: coinbase,
      _topic: 'religion',
      _description: 'Can God create a stone so heavy that He cannot lift it?',
      _extraInfo: JSON.stringify({
        resolutionSource: 'stars',
        tags: ['yo', 'mate'],
        longDescription: 'meh',
      }),
      meta: account,
      tx: {
        from: '0x44291b3c469806e625500a6184a045d2a5994058',
        to: addresses.Universe,
        value: Math.pow(10, 18),
      },
      onSent: () =>
        console.log('Market creation TX has been sent to the network'),
    },
  );

  return {
    network,
    market: result.callReturn,
    transaction: result.hash,
  };
}

// start Augur inside JSDOM to ensure clean shutdown
async function runAugurInSandbox(func: *, params: *): Promise<*> {
  const virtualConsole = new VirtualConsole();
  virtualConsole.sendTo(console);
  const dom = new JSDOM('', { runScripts: 'outside-only', virtualConsole });
  try {
    dom.window.AUGUR_SANDBOX_RUNNER_FUNC = func;
    dom.window.AUGUR_SANDBOX_RUNNER_PARAMS = params;

    const augurjs = await new Promise((resolve, reject) =>
      fs.readFile(
        'dev-artifacts/augur_sandbox_runner.js',
        { encoding: 'utf-8' },
        (err, data: string) => (err != null ? reject(err) : resolve(data)),
      ),
    );

    dom.window.eval(augurjs);

    return await nullthrows(dom.window.AUGUR_SANDBOX_RUNNER_PROMISE);
  } finally {
    dom.window.close();
  }
}

export default create_test_markets;
