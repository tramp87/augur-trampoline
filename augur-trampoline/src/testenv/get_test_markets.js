// @flow

/**
 * When we are in browser in debug mode, we want to display some
 * markets to do manual tests.
 *
 * Here we choose those markets for given network ID.
 */

import { Map as ImmMap } from 'immutable';
import invariant from 'invariant';
import { getContractAddresses } from './env';
import MARKETS_ON_PUBLIC_NETWORKS from './markets_on_public_networks';

export type TestMarket = {|
  market: string,
  transaction: string,
|};

async function get_test_markets(
  network: string,
): Promise<ImmMap<string, TestMarket>> {
  if (MARKETS_ON_PUBLIC_NETWORKS[network] != null) {
    return ImmMap(MARKETS_ON_PUBLIC_NETWORKS[network]);
  }

  console.log(`Assuming ${network} is a local network.`);
  return await get_test_markets_for_local_network(network);
}

async function get_test_markets_for_local_network(
  network: string,
): Promise<ImmMap<string, TestMarket>> {
  const json = await fetch('/api/test_markets').then(response =>
    response.json(),
  );
  invariant(json.network === network, 'Network mismatch');
  return ImmMap(json.markets);
}

async function get_local_test_network(): Promise<string> {
  const addresses = await getContractAddresses();
  return addresses.network;
}

export { get_test_markets, get_local_test_network };
