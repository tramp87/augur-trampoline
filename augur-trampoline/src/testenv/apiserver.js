// @flow

/*
 * Used to have API access to test Ethereum RPC, and to addresses of test
 * Augur contracts.
 */

import express from 'express';
import proxy from 'http-proxy-middleware';
import { TESTRPC_HTTP_URL, AUGUR_CONTRACTS_URL } from './env';
import create_test_markets from './create_test_markets';

const testMarketsPromise = create_test_markets();

const app = express();
const port = 4000;

const ethProxy = proxy('/api/eth', {
  target: TESTRPC_HTTP_URL,
  pathRewrite: {
    '^/api/eth': '',
  },
});

app.use('/api/eth', ethProxy);

const contractsProxy = proxy('/api/contracts', {
  target: AUGUR_CONTRACTS_URL,
  pathRewrite: {
    '^/api/contracts': '',
  },
});

app.use('/api/contracts', contractsProxy);

app.get('/api/test_markets', async (req, res) => {
  const testMarkets = await testMarketsPromise;
  res.send(JSON.stringify(testMarkets));
});

process
  .on('unhandledRejection', (reason, p) => {
    console.error(reason, 'Unhandled Rejection at Promise', p);
    process.exit(1);
  })
  .on('uncaughtException', err => {
    console.error(err, 'Uncaught Exception thrown');
    process.exit(1);
  });

app.listen(port, () => console.log(`API listening on port ${port}!`));
