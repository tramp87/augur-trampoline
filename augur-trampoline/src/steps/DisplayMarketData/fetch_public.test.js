// @flow

import { Map as ImmMap } from 'immutable';
import Web3 from 'web3';
import { get_test_markets } from '../../testenv/get_test_markets';
import fetchMarketData from './fetch';

const networks = ImmMap({
  '1': 'mainnet',
  '4': 'rinkeby',
});

describe('can fetch market data', () => {
  networks
    .entrySeq()
    .sort()
    .forEach(([networkID, networkName]) => {
      test(
        `from network ${networkName}`,
        async () => {
          const markets = await get_test_markets(networkID);
          const web3 = new Web3(
            new Web3.providers.HttpProvider(
              `https://${networkName}.infura.io/augur`,
            ),
          );

          const fetchedData = await Promise.all(
            markets
              .map(market =>
                fetchMarketData(web3, market.market, market.transaction),
              )
              .entrySeq()
              .map(async ([name, promise]) => {
                const data = await promise;
                return [name, data];
              })
              .toArray(),
          ).then(pairs => ImmMap(pairs));

          expect(
            fetchedData.map(datum => ({
              ...datum,
              isFinalized: '<OMITTED to make test not time-dependent>',
            })),
          ).toMatchSnapshot();
        },
        60000,
      );
    });
});
