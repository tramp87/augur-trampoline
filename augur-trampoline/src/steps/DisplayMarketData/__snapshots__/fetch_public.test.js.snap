// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can fetch market data from network mainnet 1`] = `
Immutable.Map {
  eth500: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "Will price of Ethereum exceed $500 at the end of 2018?",
    "endTime": "1546333200",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "As reported by Coinbase.pro (last trade in 2018)",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "https://pro.coinbase.com/trade/ETH-USD",
    "scalarDenomination": undefined,
  },
  rep32: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "Will REP token trade above $32 at the end of 2018?",
    "endTime": "1546333200",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "As reported by major exchanges.",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "https://coinmarketcap.com/currencies/augur/",
    "scalarDenomination": undefined,
  },
  btc20k: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "Will the Bitcoin (BTC) price exceeds $ 20 000 at midnight UTC on December 31, 2018 ?",
    "endTime": "1546300800",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "Only the bitcoin (BTC) price at midnight UTC on December 31, 2018 will be considered for the determination of the final outcome, even if the price of bitcoin exceeds USD 20 000 before the end of 2018. 
  In the event of disappearance of the site \\"coinmarketcap\\", the reporters will be able to use any other websites to give the result and to make a consensus on the final outcome.",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "https://coinmarketcap.com/",
    "scalarDenomination": undefined,
  },
  eth_options_categorical: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "Ethereum Call/Put Options, Hedging: Dec. 31, 2018.",
    "endTime": "1546477200",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "The winning contract shall be the bracket such that the last price of Ethereum on the end date exceeds the value in the bracket, without exceeding any value above it in the list of contracts.
  For example, if the end price is 499.99, the \\"400\\" bracket wins out. If end price is 12,345.67, then the top-most bracket wins out.",
    "marketCreationFee": "10000000000000000",
    "marketType": "1",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "8",
    "outcomes": Array [
      "800",
      "700",
      "600",
      "500",
      "400",
      "300",
      "200",
      "0",
    ],
    "resolutionSource": "https://pro.coinbase.com/trade/ETH-USD",
    "scalarDenomination": undefined,
  },
  trump_reelection_2020: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "Will Donald Trump be Re-Elected in 2020?",
    "endTime": "1607220000",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "Donald Trump shall be re-elected as president of the United States in 2020, as determined by the Electoral College count, regardless of any electoral college members changing their vote. Inauguration shall have no effect on this market.",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "",
    "scalarDenomination": undefined,
  },
  ethcall500: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "ETH call option at $500 [expiring 2019-01-01]",
    "endTime": "1546304400",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "If your client does not render the details properly, you can view https://gateway.ipfs.io/ipfs/QmbXL4YifbyNhKcdfBG9GAt7Bq291ZBzBWW91rYXa61p5T
  
  YES share represents European call option on ETH with strike price $500.
  Holder of NO share becomes option writer.
  
  At the expiration holder of YES share receives the amount by which ETH exceeds price of $500.
  Holder of NO share received $500 or 1 ETH, whichever is smaller.
  
  Now you can trade Ethereum volatility on Augur :)
  
  ===
  
  Instruction for reporters:
  
  Let E be price of last trade on Coinbase Pro for pair ETH/USD in year 2018 (UTC).
  
  Calculate V = (E - $500) / E.
  
  If V < 0, resolve market as 0.
  Otherwise resolve market as V, rounded to the nearest market tick.
  
  In the unlikely event that V falls exactly in the middle between two market ticks, round up.",
    "marketCreationFee": "10000000000000000",
    "marketType": "2",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "100",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "https://pro.coinbase.com/trade/ETH-USD",
    "scalarDenomination": "ETH",
  },
  usd100scalar: Object {
    "denominationToken": "0xd5524179cb7ae012f5b642c1d6d700bbaa76b96b",
    "description": "What will the price of 100 USD be at the end of 2018, denominated in ETH?",
    "endTime": "1546347600",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "The price of USD denominated in ETH is equal to 100 divided by the price of ETH in USD. For example, if the price of 1 ETH is 500 USD, then this contract settles with a value 0.2. In this case, someone with a YES share would be paid 0.2 ETH, while someone with a NO share would be paid 0.8 ETH. If the price of ETH is 100 USD, this contract settles with a value of 1, paying out 1 ETH to YES holders and 0 to NO holders. If the price of ETH is less than 100 USD, this contract will still pay out 1 ETH to YES holders and 0 to NO holders.",
    "marketCreationFee": "10000000000000000",
    "marketType": "2",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "100",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "https://pro.coinbase.com/trade/ETH-USD",
    "scalarDenomination": "ETH",
  },
}
`;

exports[`can fetch market data from network rinkeby 1`] = `
Immutable.Map {
  cfk_jail: Object {
    "denominationToken": "0x2da4d465978981bd75bbac4c9f3bda10be0b465c",
    "description": "Will the ex-president of Argentina, CFK, go to jail before the end of the year?",
    "endTime": "1546311540",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "If CFK ends up in jail before the end of the year, the premise closes at the moment of the fact.
  ONLY AN EFFECTIVE PRISON WILL BE ADMITTED, NOT A PREVENTIVE PRISON.",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "https://www.clarin.com/",
    "scalarDenomination": undefined,
  },
  trump_impeachment: Object {
    "denominationToken": "0x2da4d465978981bd75bbac4c9f3bda10be0b465c",
    "description": "Will Donald Trump be impeached by December 25th, 2018?",
    "endTime": "1545786000",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "Do you think Trump will remain president or be impeached? Put your money where your mouth is and bet on the outcome. ",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "",
    "scalarDenomination": undefined,
  },
  china_antibiotics: Object {
    "denominationToken": "0x2da4d465978981bd75bbac4c9f3bda10be0b465c",
    "description": "Will antibiotics be outlawed for agricultural use in China by the end of 2019?",
    "endTime": "1577865600",
    "isFinalized": "<OMITTED to make test not time-dependent>",
    "longDescription": "Will antibiotic pan-resistance lead to a massive resurgence of infectious diseases?",
    "marketCreationFee": "10000000000000000",
    "marketType": "0",
    "maxPrice": "1000000000000000000",
    "minPrice": "0",
    "numTicks": "10000",
    "numberOfOutcomes": "2",
    "outcomes": Array [],
    "resolutionSource": "",
    "scalarDenomination": undefined,
  },
}
`;
